python/pyarrow/tests/test_cuda_numba_interop.py

```diff
@@ -0,0 +1,124 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+import pyarrow as pa
+import numpy as np
+
+cuda = pytest.importorskip("pyarrow.cuda")
+nb_cuda = pytest.importorskip("numba.cuda")
+
+
+def make_random_buffer(size, target='host', ctx=None):
+    """Return a host or device buffer with random data.
+    """
+    if target == 'host':
+        assert size >= 0
+        buf = pa.allocate_buffer(size)
+        arr = np.frombuffer(buf, dtype=np.uint8)
+        arr[:] = np.random.randint(low=0, high=255, size=size, dtype=np.uint8)
+        return arr, buf
+    elif target == 'device':
+        arr, buf = make_random_buffer(size, target='host')
+        dbuf = ctx.new_buffer(size)
+        dbuf.copy_from_host(buf, position=0, nbytes=size)
+        return arr, dbuf
+    raise ValueError('invalid target value')
+
+
+def test_numba_memalloc():
+    from numba.cuda.cudadrv.devicearray import DeviceNDArray
+
+    # Create context instances
+    ctx = cuda.Context()
+    nb_ctx = ctx.to_numba()
+    assert ctx.handle == nb_ctx.handle.value
+    assert ctx.handle == nb_cuda.cudadrv.driver.driver.get_context().value
```

Comment:
I'm not sure what this means. Is this the default Numba Cuda context? Perhaps there is a better API to get it, such as https://numba.pydata.org/numba-doc/dev/cuda-reference/host.html#numba.cuda.current_context